jax/_src/scipy/spatial/transform.py

```diff
@@ -0,0 +1,393 @@
+# Copyright 2023 The JAX Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import functools
+import re
+import typing
+
+import scipy.spatial.transform
+
+import jax
+import jax.numpy as jnp
+from jax._src.numpy.util import _wraps
+
+
+@_wraps(scipy.spatial.transform.Rotation)
+class Rotation(typing.NamedTuple):
+  """Rotation in 3 dimensions."""
+
+  quat: jax.Array
+
+  # TODO: concatenate is not available in the oldest supported scipy version
+  # @classmethod
+  # def concatenate(cls, rotations: typing.Sequence):
+  #   """Concatenate a sequence of `Rotation` objects."""
+  #   return cls(jnp.concatenate([rotation.quat for rotation in rotations]))
```

Comment:
We should un-comment this to make the API compatible with newer scipy versions.

The problem is not that this is defined; the problem is in the test: we we need to check for the appropriate scipy version before attempting to call `concatenate` on the scipy version of this class.

Reply:
Ok, got this working.